messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -131,10 +133,37 @@ protected void initializeConfig(ClientConfig config) throws IOException {
         + " streamType:" + streamType);
   }
 
+  private void startNewCollectorDiscovererTimer(ClientConfig config) {
+       /*
+    This timer is responsible for identifying new
+    collector output files and initialising partition readers for them
+     */
+    int initialDelay = config.getInteger(initialDelayForDiscoverer,
+        DEFAULT_INITIAL_DELAY_FOR_DISCOVERER);
+    int discoverFrequency = config.getInteger(frequencyForDiscoverer,
+        DEFAULT_FREQUENCY_FOR_DISCOVERER);
+    partitionDiscovererAndReaderCreator = new Timer(
+        "PartitionDiscovererAndReaderCreator");
+    partitionDiscovererAndReaderCreator.schedule(new TimerTask() {
+      @Override
+      public void run() {
+        try {
+          if (initDone) {
```

Comment:
will be good to log some message that can be used to debug when discoverer ran, and whether it did anything.